### DISCUSSION TITLE
Add to `~/.Brewfile` on install

### DISCUSSION THREAD

**Discussion by johndbritton** [Homebrew/discussions#163] [Everyday usage] [answered]
I have a global `~/.Brewfile` that I use to manage everything installed on my Mac.

Whenever I install something new, it usually goes something like this:

1. Find an app I want to install
2. `brew search name`
3. `brew install name`
4. edit `~/.Brewfile` to add the app

Is there a way to do this in one step, something like `brew install name --save` or `brew bundle add name`? I've searched around but haven't found anything that does what I'm looking for. If it doesn't exist, where would be the best place to send a PR `brew` or `homebrew-bundle`? Is this something that you'd accept a pull request for?

**Comment by Rylan12** [comment, marked as answer]
One thing that may make your life easier: `brew bundle dump --global` will automatically write all your currently installed formulae/casks/etc. to `~/.Brewfile`, saving you the manual edit step (assuming you want all of your currently installed formulae in the brewfile).

If that's not what you're looking for, I could see a `brew bundle add <formula>` and `brew bundle remove <formula>` being useful. This would belong in the [Homebrew/homebrew-bundle](https://github.com/Homebrew/homebrew-bundle) repo. If you want feedback on the idea before spending time implementing it, you should open an issue there to get maintainer feedback (not sure how many of the folks who work on `brew bundle` follow these discussions so you'll probably get better feedback there)

Since the `brew bundle` functionality is in a separate tap that users may not have, I think adding a flag to `brew install` probably is a no-go.

Hope that's helpful

**Reply by gromgit** [reply]
> One thing that may make your life easier: `brew bundle dump --global`

Yeah, that's in my crontab on every Homebrew box (Linux and macOS), scheduled to run 5 minutes before the nightly backup. @johndbritton , unless you need up-to-the-minute Brewfiles, this might be a more practical way to get what you want.

**Reply by johndbritton** [reply]
I'm aware of `brew bundle dump --global` but I don't necessarily want every installed package to be in my `Brewfile`. I'll open an issue to discuss on https://github.com/Homebrew/homebrew-bundle.

**Reply by johndbritton** [reply]
Opened an issue here: https://github.com/Homebrew/homebrew-bundle/issues/818
